components/eventsrv/src/client.rs

```diff
@@ -32,23 +33,30 @@ use std::io::prelude::*;
 use std::path::Path;
 use std::thread::sleep;
 use std::time::Duration;
-
+use rand::{thread_rng, Rng};
 use zmq::{Context, PUSH};
 use protobuf::Message;
 
 use message::event::{EventEnvelope, EventEnvelope_Type};
 
 fn main() {
+
+    let mut args: Vec<_> = env::args().collect();
+    args.remove(0); // drop the binary name
+
+    let file_arg = args.remove(0);
+
     let ctx = Context::new();
     let socket = ctx.socket(PUSH).unwrap();
-    assert!(socket.bind("tcp://*:34567").is_ok());
 
-    let arg = match env::args().last() {
-        Some(a) => a,
-        None => panic!("Pass the path for the file to parse"),
-    };
+    // Everything else is treated as a port number
+    for p in args {
+        let push_connect = format!("tcp://localhost:{}", p);
+        println!("connecting to {}", push_connect);
+        assert!(socket.connect(&push_connect).is_ok());
```

Comment:
Just for my own edification - we changed this from `bind` to `connect` because of @reset's comment about the "stable" parts of the system should `bind` and the transient parts `connect`.  Is that right?